pydra/compose/base/helpers.py

```diff
@@ -183,7 +183,7 @@ def extract_function_inputs_and_outputs(
         input_types[p.name] = type_hints.get(p.name, ty.Any)
         if p.default is not inspect.Parameter.empty:
             input_defaults[p.name] = p.default
-    if inputs:
+    if inputs is not None:
         if not isinstance(inputs, dict):
             raise ValueError(
                 f"Input names ({inputs}) should not be provided when "
```

Comment:
Just looking at this, it seems like `inputs` could be a `list[str | Arg]`, but it isn't turned into a dict before this. This means we'll now error on an empty list as well as a populated list, but maybe we should just remove the list annotation?

Reply:
Now test for list[Arg] and convert to dict[str, Arg] if they are named (raise an error otherwise)